main.js

```diff
@@ -0,0 +1,91 @@
+const toDoList = [ 
+	{name: 'погулять', status: 'In progress', priority: 'low'}, 
+    {name: 'сходить на рыбалку', status: 'Done', priority: 'high'},
+    {name: 'почитать', status: 'In progress', priority: 'high'}
+];
+
+const STATUSES = [ "To Do", "In progress", "Done" ]
+const PRIORITY = [ 'low', 'high' ]
+
+function checkName(name) {
+    let checkName = toDoList.find(designation => designation.name === name)
+    return checkName
+};
+function checkStatuses(status) {
+    let checkStatuses = STATUSES.find(state => state === status)
+    return checkStatuses
+};
+function checkPriority(priority) {
+    let checkPriority = PRIORITY.find(importance => importance === priority)
+    return checkPriority
+}
+
+const MASSGES = {
+    REPEAT_ERROR: "Эта задача уже есть в списке",
+    STATUS_ERROR: "Введите верный статус",
+    PRIORITY_ERROR: "Введите верный приоритет",
+    ADDING_ERROR: "Нельзя удалить того чего нет",
+    STATUS_CHANGE_ERROR: "Нельзя поменять статус того чего нет",
+}
+
+
+function addTask(name, status, priority) {
+    if (checkName(name)) {
+        console.log(MASSGES.REPEAT_ERROR)
+    } else if (!checkStatuses(status)) {
+        console.log(MASSGES.STATUS_ERROR)
+    } else if (!checkPriority(priority)) {
+        console.log(MASSGES.PRIORITY_ERROR)
+    } else {
+        toDoList.push({name, status, priority})
+    };
+}
+
+function deleteTask(name) {
+    if (checkName(name)) {
+        const numberDeleteTask = toDoList.findIndex(task => task.name === name)
+        toDoList.splice(numberDeleteTask, 1)
+    } else {
+        console.log(MASSGES.ADDING_ERROR)
+    };
+}
+
+function changeStatus(name, status) {
+    if (checkName(name)) {
+        toDoList.forEach(task => {
+            if (task.name === name) {
+                task.status = status
+            }
+        });
+    } else {
+        console.log(MASSGES.STATUS_CHANGE_ERROR)
+    };
+}
+
+function showList() {
+    let counter = false
+
+    STATUSES.forEach(condition => {
+        console.log(condition + ":")
+
+        toDoList.forEach(task => {
+            if (task.status === condition) {
+            console.log(`\t${task.name} , приоритетность: ${task.priority}`)
+            counter = true
+            }
+        })
+        if (counter === false) {
+            console.log("\t-")
+        }
+    })
+}
```

Comment:
Я бы сделал отсутствие задач состоянием по умолчанию, а при нахождении менял на false
```suggestion
function showList() {
    let noTasks = true

    STATUSES.forEach(condition => {
        console.log(condition + ":")

        toDoList.forEach(task => {
            if (task.status === condition) {
            console.log(`\t${task.name} , приоритетность: ${task.priority}`)
            noTasks = false
            }
        })
        if (noTasks) {
            console.log("\t-")
        }
    })
}
```